packages/boot/src/mixins/boot.mixin.ts

```diff
@@ -71,6 +71,23 @@ export function BootMixin<T extends MixinTarget<Application>>(superClass: T) {
       );
     }
 
+    booted: boolean;
+
+    /**
+     * Override to detect and warn about starting without booting.
+     */
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    public async start(): Promise<void> {
+      await super.start();
+      if (!this.booted) {
+        console.warn(
```

Comment:
I wish we have a better way for warnings. `console.warn` is not very flexible.

Reply:
Let's use `process.emitWarning()` - https://nodejs.org/api/process.html#process_process_emitwarning_warning_type_code_ctor

Reply:
Wouldn't the warning be lost if they are not being actively listened for? If it requires an additional step for the developer to be able to be warned of problems, it is not en effective solution. The `console.log()` method is also used by [datasource](https://github.com/strongloop/loopback-next/blob/master/packages/boot/src/booters/datasource.booter.ts#L56) and [repository](https://github.com/strongloop/loopback-next/blob/master/packages/boot/src/booters/repository.booter.ts#L52) booters.

If `console.log()` is not flexible enough, we should be come up with a solution for it instead of passing on the pain to the user.

Reply:
`process.emitWarning(...)` will print the message to `stderr` as I tested:

```
process.emitWarning('xyz');
> (node:85982) Warning: xyz
```

Reply:
I see, updated.